botocore/exceptions.py

```diff
@@ -305,3 +305,18 @@ def __init__(self, error_response, operation_name):
 
 class ImminentRemovalWarning(Warning):
     pass
+
+
+class InvalidDNSNameError(BotoCoreError):
+    """Error when virtual host path is forced on a non-DNS compatible bucket"""
+    fmt = (
+        'Bucket named {bucket_name} is not DNS compatible. Virtual '
+        'hosted-style access cannot be used.'
```

Comment:
Should we provide a pointer to not configuring your client to use virtual hosted buckets?